File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string excerpt);
int count_words(string excerpt);
int count_sentences(string excerpt);

int main(void)
{
    string excerpt = get_string("Your Excerpt: \n");

    float L = ((float) count_letters(excerpt) / (float) count_words(excerpt)) * 100.0;

    float S = (float) count_sentences(excerpt) / (float) count_words(excerpt) * 100.0;

    float index = 0.0588 * L - 0.296 * S - 15.8;

    int index_rounded = round(index);s

    if (index_rounded < 16 && index_rounded > 0)
    {
        printf("Grade %i\n", index_rounded);
    }
    else if (index_rounded < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }


}

int count_letters(string excerpt)
{
    int letters = 0;
    for (int i = 0; i < strlen(excerpt); i++)
    {
        if (isalpha(excerpt[i]))
        {
            letters++;
        }
    }

    return letters;
}

int count_words(string excerpt)
{
    int words = 0;
    for (int i = 0; i <= strlen(excerpt); i++)
    {
        if (isspace(excerpt[i]))
        {
            words++;
        }
    }

    return words + 1;
}

int count_sentences(string excerpt)
{
    int sentences = 0;
    for (int i = 0; i < strlen(excerpt); i++)
    {
        if (excerpt[i] == '!' || excerpt[i] == '.' || excerpt[i] == '?')
        {
            sentences++;
        }
    }

    return sentences;
}

File: sentimental-readability/readability.py
import cs50


def main():

    excerpt = cs50.get_string("Your excerpt: ")

    L = count_letters(excerpt) / count_words(excerpt) * 100

    S = count_sentences(excerpt) / count_words(excerpt) * 100

    index = 0.0588 * L - 0.296 * S - 15.8

    index_rounded = round(index)

    if (index_rounded < 16 and index_rounded > 0):
        print(f"Grade {index_rounded}")

    elif (index_rounded < 1):
        print("Before Grade 1")

    else:
        print("Grade 16+")


def count_letters(excerpt):

    letters = 0

    for i in range(len(excerpt)):
        if str.isalpha(excerpt[i]):
            letters += 1

    return letters


def count_words(excerpt):

    words = 0

    for i in range(len(excerpt)):
        if str.isspace(excerpt[i]):
            words += 1

    return words + 1


def count_sentences(excerpt):

    sentences = 0

    for i in range(len(excerpt)):
        if excerpt[i] == '!' or excerpt[i] == '.' or excerpt[i] == '?':
            sentences += 1

    return sentences


if __name__ == "__main__":
    main()